pkg/rancher/rancher.go

```diff
@@ -258,6 +258,9 @@ func New(ctx context.Context, clientConfg clientcmd.ClientConfig, opts *Options)
 }
 
 func (r *Rancher) Start(ctx context.Context) error {
+	if err := dashboardapi.Add(ctx, r.Wrangler); err != nil {
```

Comment:
Putting this here means every rancher pod in the cluster will attempt to add the cluster repos. This can cause a race condition where two pods try to create or update the resource simultaneously, which would cause one of them to fail and crash the pod.

Reply:
The race condition could be addressed by checking the errors of create and update and then retrying if they are conflict errors.
For example if you attempt to create a repo and get already exists error we can do another get and verify that it has the expected values.